File: src/Application.Resources/scripts/event/AreaBossKingSageCat.js
function init() {
    scheduleNew();
}

function scheduleNew() {
    setupTask = em.schedule("start", 0);    //spawns upon server start. Each 3 hours an server event checks if boss exists, if not spawns it instantly.
}

function cancelSchedule() {
    if (setupTask != null) {
        setupTask.cancel(true);
    }
}

function start() {
    var goblinForest2 = em.getChannelServer().getMapFactory().getMap(250010504);
    var kingSageCat = LifeFactory.getMonster(7220002);

    if (goblinForest2.getMonsterById(7220002) != null) {
        em.schedule("start", 3 * 60 * 60 * 1000);
        return;
    }
    var posX;
    var posY = 540;
    posX = Math.floor((Math.random() * 1300) - 500);
    const spawnpoint = new Point(posX, posY);
    goblinForest2.spawnMonsterOnGroundBelow(kingSageCat, spawnpoint);
    goblinForest2.broadcastMessage(PacketCreator.serverNotice(6, "The ghostly air around here has become stronger. The unpleasant sound of a cat crying can be heard."));
    em.schedule("start", 3 * 60 * 60 * 1000);
}

// ---------- FILLER FUNCTIONS ----------

function dispose() {}

function setup(eim, leaderid) {}

function monsterValue(eim, mobid) {return 0;}

function disbandParty(eim, player) {}

function playerDisconnected(eim, player) {}

function playerEntry(eim, player) {}

function monsterKilled(mob, eim) {}

function scheduledTimeout(eim) {}

function afterSetup(eim) {}

function changedLeader(eim, leader) {}

function playerExit(eim, player) {}

function leftParty(eim, player) {}

function clearPQ(eim) {}

function allMonstersDead(eim) {}

function playerUnregistered(eim, player) {}

File: src/Application.Resources/scripts/event-lua/AreaBossKingSageCat.lua
local Common_AreaBoss = require("scripts/event-lua/__BaseAreaBoss")

-- 继承基类方法
for k, v in pairs(Common_AreaBoss) do
    _ENV[k] = v
end

-- 重写 init 方法
function init()
    scheduleNew()
end

-- 重写 start 方法
function start()
    local posX = math.floor(math.random(1300) - 500)
    local posY = 540
    Common_AreaBoss.spawnBoss(
        250010504,          -- 地图ID
        7220002,            -- 怪物ID 妖怪禅师
        posX,               -- X坐标
        posY,               -- Y坐标
        "周围的鬼气更加浓重了。传来一阵令人不快的猫叫声。"  -- 提示消息
    )
end